comps/cores/mega/gateway.py

```diff
@@ -859,16 +880,47 @@ async def handle_request(self, request: Request):
             stream_opt = False
         chat_request = ChatCompletionRequest.model_validate(data)
         # Multimodal RAG QnA With Videos has not yet accepts image as input during QnA.
-        prompt_and_image = self._handle_message(chat_request.messages)
-        if isinstance(prompt_and_image, tuple):
+        messages = self._handle_message(chat_request.messages)
+        if isinstance(messages, tuple):
             # print(f"This request include image, thus it is a follow-up query. Using lvm megaservice")
-            prompt, images = prompt_and_image
+            prompt, b64_types = messages
+            decoded_audio_input = ""
+            if "audio" in b64_types:
+                # call ASR endpoint to decode audio to text             
+                input_dict = {"byte_str": b64_types["audio"][0]}
+                response = requests.post(self.asr_endpoint, data=json.dumps(input_dict), proxies={"http": None})
```

Comment:
Since we are doing this same thing in a couple of places, maybe create an audio to text function to reduce duplication.